### PR TITLE
[ADD] Add state module

### PR DESCRIPTION
**Pull request by lfreeke**

### CODE REVIEW

**Comment by lfreeke** [comment]
@NL66278 Ping

File: l10n_nl_state/__manifest__.py
Comment (NL66278) [inline l10n_nl_state/__manifest__.py]: make it https:

File: l10n_nl_state/data/l10n_nl_states.xml
Comment (NL66278) [inline l10n_nl_state/data/l10n_nl_states.xml]: Hoi sluit voor de codes e.d. aan bij de ISO standaard: https://en.wikipedia.org/wiki/ISO_3166-2:NL (doe die link ook maar in de README.rst).

En neem dan gelijk even de zes eilanden en hun codes mee.

**Review by NL66278 — APPROVED** [review]
Checked with standard and LGTM

File: l10n_nl_state/__manifest__.py
Comment (NL66278) [inline l10n_nl_state/__manifest__.py]: Line 17 and 18 are missing one space, so not aligned with rest. After improving this, squash all commits into one.

File: l10n_nl_state/README.rst
Comment (hbrunn) [inline l10n_nl_state/README.rst]: use a human readable name here, and use the same in the `name` key of the manifest. And make this an rst title by have the same amount of equal signs as text

File: l10n_nl_state/README.rst
Comment (hbrunn) [inline l10n_nl_state/README.rst]: delete

File: l10n_nl_state/README.rst
Comment (hbrunn) [inline l10n_nl_state/README.rst]: this sounds very weird, rephrase

File: l10n_nl_state/README.rst
Comment (hbrunn) [inline l10n_nl_state/README.rst]: delete and resolve `repo_id` above

File: l10n_nl_state/README.rst
Comment (hbrunn) [inline l10n_nl_state/README.rst]: remove if there are none

File: l10n_nl_state/__manifest__.py
Comment (hbrunn) [inline l10n_nl_state/__manifest__.py]: remove or add summary

File: l10n_nl_state/__manifest__.py
Comment (hbrunn) [inline l10n_nl_state/__manifest__.py]: use a human readable name

**Review by hbrunn — CHANGES_REQUESTED** [review]

**Comment by hbrunn** [comment]
@lfreeke thanks for the update, but we seem to have two modules in the PR now, can you check?

I'm also discussing an extension for this with @NL66278 to have this module also know which postcodes belong to which province, so that we can autofill the province in an onchange of the zip code

**Comment by lfreeke** [comment]
@hbrunn Oh let me check it out. I renamed the module but something must have go wrong. 
Maybe we can have a look at this. https://github.com/OCA/partner-contact/tree/10.0/base_location

**Comment by lfreeke** [comment]
@hbrunn it is fixed now.

**Review by hbrunn — APPROVED** [review]

**Review by NL66278 — APPROVED** [review]
LGTM and tested

**Review by hbrunn — CHANGES_REQUESTED** [review]
@lfreeke I need to retract my approval: Use the xmlid if the country, not some database id

**Comment by lfreeke** [comment]
@hbrunn I change the database id for xmlid

**Review by hbrunn — APPROVED** [review]
thanks, I also added https://github.com/lfreeke/l10n-netherlands/pull/1 to have this look up the province from the zip code. So if you merge this, we can merge the whole thing if I didn't mess up the tests

**Comment by NL66278** [comment]
@hbrunn I cannot merge this. I have no merge authority in the oca anywhere,

**Comment by lfreeke** [comment]
@NL66278 I merged it into my branch.

**Comment by hbrunn** [comment]
...and lfreeke#2 for style missers

@NL66278 you can apply as a member of https://github.com/orgs/OCA/teams/local-netherlands-maintainers (and some others too I guess) to get write permissions. People will want to see links to contributions in the appropriate repo for this to happen

File: l10n_nl_country_states/models/res_country_state_nl_zip.py
Comment (NL66278) [inline l10n_nl_country_states/models/res_country_state_nl_zip.py]: Use copyright as text (instead of sign) and https: //therp.nl (ssl) everywhere.

File: l10n_nl_country_states/__scripts__/generate_postcode_regios.py
Comment (NL66278) [inline l10n_nl_country_states/__scripts__/generate_postcode_regios.py:1]: Would it be possible to sort the excell sheel by zip code?

Then you could safe the province (res.country.state), min_zip encountered and max_zip encounterd until province changes. And only then write range. This would I think significantly reduce the resulting file in size, while still supporting having several ranges within one province (and a range laying between in another province).

Comment (hbrunn) [reply l10n_nl_country_states/__scripts__/generate_postcode_regios.py:1]: sure, that's the whole point of the data model. Just didn't come to this yet

**Pull request merged**
